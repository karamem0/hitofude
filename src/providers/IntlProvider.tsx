//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import { IntlProvider as Provider, createIntl } from 'react-intl';
import en from '../translations/en.json';
import ja from '../translations/ja.json';

const translations: { [key: string]: Record<string, string> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language]
});

function IntlProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
