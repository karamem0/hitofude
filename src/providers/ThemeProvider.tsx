//
// Copyright (c) 2023-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import {
  FluentProvider as Provider,
  Theme
} from '@fluentui/react-components';

import { darkTheme, lightTheme } from '../themes/Theme';
import { InvalidOperationError } from '../types/Error';
import { ThemeName } from '../types/Model';

interface ThemeState {
  theme: Theme,
  themeName: ThemeName
}

interface ThemeContextState {
  theme: ThemeState,
  changeTheme: (themeName: ThemeName) => void
}

const ThemeContext = React.createContext<ThemeContextState | undefined>(undefined);

export const useTheme = (): ThemeContextState => {
  const value = React.useContext(ThemeContext);
  if (value == null) {
    throw new InvalidOperationError();
  }
  return value;
};

function ThemeProvider(props: Readonly<React.PropsWithChildren<unknown>>) {

  const { children } = props;

  const [ theme, setTheme ] = React.useState<ThemeState>({
    theme: lightTheme,
    themeName: ThemeName.light
  });

  const handleChangeTheme = React.useMemo(() => (value: ThemeName) => {
    switch (value) {
      case ThemeName.light:
        setTheme(() => ({
          theme: lightTheme,
          themeName: value
        }));
        break;
      case ThemeName.dark:
        setTheme(() => ({
          theme: darkTheme,
          themeName: value
        }));
        break;
      default:
        break;
    }
  }, []);

  const value = React.useMemo<ThemeContextState>(() => ({
    theme,
    changeTheme: handleChangeTheme
  }), [
    theme,
    handleChangeTheme
  ]);

  return (
    <ThemeContext.Provider value={value}>
      <Provider theme={value.theme.theme}>
        {children}
      </Provider>
    </ThemeContext.Provider>
  );

}

export default ThemeProvider;
