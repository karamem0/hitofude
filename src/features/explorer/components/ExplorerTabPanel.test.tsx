//
// Copyright (c) 2023-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/hitofude/blob/main/LICENSE
//

import React from 'react';

import IntlProvider from '../../../providers/IntlProvider';
import Presenter from './ExplorerTabPanel.presenter';
import ThemeProvider from '../../../providers/ThemeProvider';
import { render } from '@testing-library/react';

vi.mock('./ExplorerFileTreeItem', () => ({
  default: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="test-ExplorerFileTreeItem">
      {children}
    </div>
  )
}));

vi.mock('./ExplorerFolderTreeItem', () => ({
  default: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="test-ExplorerFolderTreeItem">
      {children}
    </div>
  )
}));

vi.mock('../../../common/components/TreeHeader', () => ({
  default: ({ children }: React.PropsWithChildren<unknown>) => (
    <div data-testid="test-TreeHeader">
      {children}
    </div>
  )
}));

it('should match the snapshot when the folder is not undedined', () => {
  const params = {
    allFiles: true,
    fileConflicts: [],
    selectedFolder: {
      id: '01BYE5RZ6QN3ZWBTUFOFD3GSPGOHDJD36K',
      name: 'Attachments',
      parentId: '01BYE5RZ56Y2GOVW7725BZO354PWSELRRZ',
      files: [
        {
          id: '01BYE5RZ4FL7NVUU4UDVHIRG32OLTW4IKM',
          baseName: 'Annual Financial Report (DRAFT)',
          fullName: 'Annual Financial Report (DRAFT).docx',
          extension: '.docx'
        }
      ]
    }
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it('should match the snapshot when the folder is undedined', () => {
  const params = {
    allFiles: true,
    fileConflicts: [],
    selectedFolder: undefined
  };
  const { asFragment } = render(
    <IntlProvider>
      <ThemeProvider>
        <Presenter {...params} />
      </ThemeProvider>
    </IntlProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
