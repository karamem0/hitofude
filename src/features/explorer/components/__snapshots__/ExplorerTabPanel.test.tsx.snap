// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot when the folder is not undedined 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: grid;
  grid-template-rows: 1rem 2rem calc(calc(100svh - 5.5rem) - 5.5rem) 2rem;
  grid-template-columns: 1fr;
  grid-gap: 0.5rem;
}

.emotion-1 {
  padding: 0 0.5rem;
  font-size: 12px;
  line-height: calc(16px * 1.25);
  text-transform: uppercase;
}

.emotion-2 {
  position: relative;
  display: grid;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  grid-gap: 0.25rem;
  overflow: hidden auto;
}

.emotion-4 {
  display: grid;
  padding: 0 0.5rem;
}

<div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      class="emotion-0"
    >
      <h2
        class="fui-Text emotion-1 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
      >
        Explorer
      </h2>
      <div
        data-testid="test-TreeHeader"
      />
      <div
        class="emotion-2"
        role="presentation"
        tabindex="-1"
      >
        <input
          accept="text/markdown,.md"
          multiple=""
          style="border: 0px; clip: rect(0px, 0px, 0px, 0px); clip-path: inset(50%); height: 1px; margin: 0px -1px -1px 0px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          tabindex="-1"
          type="file"
        />
        <div
          class="emotion-3"
          role="menu"
          tabindex="0"
        >
          <div
            data-testid="test-ExplorerFolderTreeItem"
          />
          <div
            data-testid="test-ExplorerFileTreeItem"
          />
        </div>
      </div>
      <div
        class="emotion-4"
      >
        <button
          aria-label="New file"
          class="fui-Button r1alrhcs ___1xubhwp_oht9j80 f1c21dwh fjxutwb fwiml72"
          title="New file"
          type="button"
        >
          <span
            class="fui-Button__icon rywnvv2 ___963sj20_1qpm2bl f1nizpg2"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2c.28 0 .5.22.5.5v5h5a.5.5 0 0 1 0 1h-5v5a.5.5 0 0 1-1 0v-5h-5a.5.5 0 0 1 0-1h5v-5c0-.28.22-.5.5-.5Z"
                fill="currentColor"
              />
            </svg>
          </span>
          New file
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should match the snapshot when the folder is undedined 1`] = `
<DocumentFragment>
  <div
    class="fui-FluentProvider fui-FluentProviderr1 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  />
</DocumentFragment>
`;
